static/app/utils/tokenizeSearch.tsx

```diff
@@ -62,6 +62,8 @@ export class MutableSearch {
 
       if (value === EMPTY_OPTION_VALUE) {
         query.addFilterValue('!has', key);
+      } else if (Array.isArray(value)) {
+        query.addFilterValues(key, value, !ALLOWED_WILDCARD_FIELDS.includes(key));
```

Comment:
I think this actually produces a bad query. I think I tried this and what I ended up getting was a condition that looked like
```
foo:val1 foo:val2 foo:val3
```
which is guaranteed to return no results since all 3 conditions cannot be true at the same time

Reply:
@Zylphrex It _can_ produce a bad query, but not necessarily! It's even in the spec:

```jsx
      results.addFilterValues('c', ['c1', 'c2']);
      expect(results.formatString()).toEqual('a:a b:b c:c1 c:c2');
```

I added this to support use cases where it produces a good query, e.g.,

```jsx
MutableSearch.fromQueryObject({
  'span.duration': ['>0', '<100'
});
```

which produces a health query: `'span.duration:>0 span.duration:<100'`. This is an existing use-case in the INP code, so I opted to support it directly.

I'm delegating down to `addFilterValues` which seems like a good fit for `string[]` values. Maybe an alternative would be to concatenate them instead and turn it into `foo:[val1,val2,val3]` which also seems natural.

Thoughts?

Reply:
Ahh, I didn't realize you can put the operators directly in the values. This like a footgun that we'll just setting ourselves up for in the future.

But because operators can be in the values, it means something like `foo:[val1,val2,val3]` is also not guaranteed to work if you take your example. So the syntax would then need to know when it should produce the list vs multiple conditions.

I don't think there's a nice solution for this if operators can be in the value but do correct me if I'm wrong.

Reply:
What kinds of guarantees can/should `fromQueryObject` provide, though? I wouldn't mind removing the `Array.isArray` bit, I'll just have to manually construct queries in a few places.

`MutableSearch` is pretty sparse on helpers to add tokens to searches, so this `fromQueryObject` factory is handy. I chose to structure it to address the way we already construct searches from objects, so I could cleanly use `MutableSearch.fromQueryObject` everywhere we pass an object around, but I can make it more strict. The strictness will reduce ergonomics, though

Reply:
I'm just trying to avoid implementing possible footguns that'll bite us back later by introducing these easy-to-encounter bugs. I'm tempted to say we should remove arrays because there's different behaviours possible depending on the key. Or something explicit to control what the resulting query will be because as it is, it's ambiguous and both use cases are valid.

Reply:
Okay, one more argument from me. I did a little more digging, and IMO the behaviour is correct as I implemented it! Here's why:

```jsx
const search = new MutableSearch(['inp:>0', 'inp:<100']);
console.log(search.filters); // { inp: [ '>0', '<100' ] }

const search2 = MutableSearch.fromQueryObject({inp: ['>0', '<100']});
console.log(search2.filters); // { inp: [ '>0', '<100' ] }
``` 

This is consistent. The way the object serializes filter tokens into an object and the way it constructs filter tokens _from_ an object are the same. Omitting the `Array.isArray` handling would make this inconsistent!

Reply:
If you're still not convinced I'll remove the `Array.isArray` handling _but_ IMO the consistently is worth it. It's not any more of a footgun than query construction being a footgun

Reply:
My main grip is that I can easily do something like

```jsx
const search = new MutableSearch.fromQuery({'http.status_code': [400, 401]});
```

and get back a bad query. But you are right that it is consistent in the sense that using the constructor would give the same consistent result so perhaps that is worth while to have.